requestmanager/asyncloader/unverifiedblockstore/unverifiedblockstore.go

```diff
@@ -55,7 +59,11 @@ func (ubs *UnverifiedBlockStore) VerifyBlock(lnk ipld.Link) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	_, err = buffer.Write(data)
+	if settable, ok := buffer.(settableWriter); ok {
```

Comment:
This may be a bit too evil, but...

```go
if buf, ok := buffer.(*bytes.Buffer); ok {
  *buf = *bytes.NewBuffer(data)
} else {
  buffer.Write(data)
}
```